src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LspEditorFeatureDetector.cs

```diff
@@ -49,7 +54,7 @@ internal sealed class LspEditorFeatureDetector(
     /// Returns <see langword="true"/> if the LSP-based editor is available.
     /// </summary>
     public bool IsLspEditorAvailable()
-        => !_useLegacyEditorLazy.Value;
+        => !_useLegacyEditorLazy.GetValue();
```

Comment:
Does this end up calling `JTF.Run` because of the async lazy?

Reply:
I believe that's right, but I would _expect_ that these would complete synchronously. Perhaps I'm wrong about how these work?

Reply:
Yeah, it looks to me like this is going to be a problem. Sigh...

https://github.com/microsoft/vs-threading/blob/2f713da7b2554e37d2473e376804e0d109503f22/src/Microsoft.VisualStudio.Threading/AsyncLazy%601.cs#L274-L289

```C#
    public T GetValue(CancellationToken cancellationToken)
    {
        // As a perf optimization, avoid calling JTF or GetValueAsync if the value factory has already completed.
        if (this.IsValueFactoryCompleted)
        {
            RoslynDebug.Assert(this.value is object);


            return this.value.GetAwaiter().GetResult();
        }
        else
        {
            return this.jobFactory is JoinableTaskFactory jtf
                ? jtf.Run(() => this.GetValueAsync(cancellationToken))
                : this.GetValueAsync(cancellationToken).GetAwaiter().GetResult();
        }
    }
```

Perhaps the right thing is just _not_ pass a `JTF` in the constructor? Or, call `GetValueAsync().GetAwaiter().GetResult()` ourselves? I would expect that any tasks would have already completed before these are accessed. So, `GetValueAsync().GetAwaiter().GetResult()` would be safe. That's an assumption on my part though.

Reply:
I did already kick off build to get a test insertion, and I'll link it here once there's an insertion PR. I definitely don't want to regress the "`JTF.Run` on background" counters. Honestly, I'm not expecting to see any `JTF.Run` changes in the test insertion. The original code backing `IsLspEditorAvailable()` likely would've had implicitly `JTF.Run` calls due to how it acquired the services. Looking at the callers, I suspect it the first call would come from `RazorFilePathToContentTypeProvider.TryGetContentTypeForFilePath(...)`, which would be called on the UI thread.

Reply:
> Looking at the callers, I suspect it the first call would come from `RazorFilePathToContentTypeProvider.TryGetContentTypeForFilePath(...)`, which would be called on the UI thread.

Actually, that would only be the case for a project opened with a file open. That might not be true in other scenarios.

Reply:
I'd suggest we either remove `JTF` from the lazy, or add a `JTF` assert for foreground thread. I agree it seems unlikely but at least then we can easily catch it if it does. The first one would result in a hang with our code on the stack very identifiably, I think the reason for the latter is just how far down the stack we are past JTF goo, so an assert/throw would be far more evident

Reply:
Yeah, that makes good sense.

Reply:
VS test insertion for the current state of this PR: https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/555670